contracts/extensions/VotingReputation.sol

```diff
@@ -0,0 +1,893 @@
+
+      emit PollEventSet(_pollId, SUBMIT_END);
+    }
+  }
+
+  /// @notice Reveal a vote secret for a poll
+  /// @param _pollId The id of the poll
+  /// @param _salt The salt used to hash the vote
+  /// @param _vote The side being supported (0 = NAY, 1 = YAY)
+  /// @param _key Reputation tree key for the staker/domain
+  /// @param _value Reputation tree value for the staker/domain
+  /// @param _branchMask The branchmask of the proof
+  /// @param _siblings The siblings of the proof
+  function revealVote(
+    uint256 _pollId,
+    bytes32 _salt,
+    uint256 _vote,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    Poll storage poll = polls[_pollId];
+    require(getPollState(_pollId) == PollState.Reveal, "voting-rep-poll-not-reveal");
+
+    uint256 userRep = getReputationFromProof(_pollId, msg.sender, _key, _value, _branchMask, _siblings);
+
+    bytes32 voteSecret = voteSecrets[_pollId][msg.sender];
+    require(voteSecret == getVoteSecret(_salt, _vote), "voting-rep-secret-no-match");
+    delete voteSecrets[_pollId][msg.sender];
+
+    poll.votes[_vote] = add(poll.votes[_vote], userRep);
+    poll.repRevealed = add(poll.repRevealed, userRep);
+
+
+    uint256 fractionUserReputation = wdiv(userRep, poll.skillRep);
+    uint256 totalStake = add(poll.stakes[YAY], poll.stakes[NAY]);
+    uint256 voterReward = wmul(wmul(fractionUserReputation, totalStake), voterRewardFraction);
+
+    poll.paidVoterComp = add(poll.paidVoterComp, voterReward);
+    tokenLocking.transfer(token, voterReward, msg.sender, true);
+
+    emit PollVoteRevealed(_pollId, msg.sender, _vote);
+
+    if (poll.repRevealed == poll.repSubmitted) {
+      poll.events[REVEAL_END] = uint64(now);
+
+      emit PollEventSet(_pollId, REVEAL_END);
+    }
+  }
+
+  /// @notice Escalate a poll to a higher domain
+  /// @param _pollId The id of the poll
+  /// @param _newDomainId The desired domain of escalation
+  /// @param _childSkillIndex For the current domain, relative to the escalated domain
+  /// @param _key Reputation tree key for the new domain
+  /// @param _value Reputation tree value for the new domain
+  /// @param _branchMask The branchmask of the proof
+  /// @param _siblings The siblings of the proof
+  function escalatePoll(
+    uint256 _pollId,
+    uint256 _newDomainId,
+    uint256 _childSkillIndex,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    Poll storage poll = polls[_pollId];
+    require(getPollState(_pollId) == PollState.Closed, "voting-rep-poll-not-closed");
+
+    uint256 newDomainSkillId = colony.getDomain(_newDomainId).skillId;
+    uint256 childSkillId = colonyNetwork.getChildSkillId(newDomainSkillId, _childSkillIndex);
+    require(childSkillId == poll.skillId, "voting-rep-invalid-domain-proof");
+
+    delete poll.events;
```

Comment:
Is this needed? Why is setting the entries on the immediately following lines not sufficient?

Reply:
Ah good point